.github/workflows/test_runner.yml

```diff
@@ -0,0 +1,161 @@
+name: Tests
+
+# This file implements the protection strategy laid out in
+# go/protobuf-gha-protected-resources.  Pull requests from branches within this
+# repository are considered safe and will immediately start running tests on
+# every commit.  Pull requests from forked repositories are unsafe, and leave
+# us vulnerable to PWN requests and stolen resources.  In these cases, we
+# require a special "safe for tests" tag to be added to the pull request before
+# we start testing.  This will be immediately removed, so that further commits
+# require their own stamp to test.
+
+on:
+  # continuous
+  schedule:
+    # Run daily at 10 AM UTC (2 AM PDT)
+    - cron: 0 10 * * *
+
+  # postsubmit
+  push:
+    branches:
+      - main
+      - '[0-9]+.x'
+      # The 21.x and 22.x branches still use Kokoro
+      - '!2[12].x'
```

Comment:
We should update all of these to not exclude 22.x now (I have no idea why the tests even ran)

Reply:
Done. I think these were initially triggered because I accidentally created the PR as a merge to main and then switched it to 22.x. 

Also, not sure why but the deleted failing kokoro tests seem to have disappeared after pushing -- no complaints there :)